### PR TITLE
Add article admin feature

### PR DESCRIPTION
**Pull request by Jexinte**
The following things have been done :

- Administration Panel is no longer accessible without been log as admin
- Admin are now able to add article 
- Logout Function 
- A Session Manager have been created allowing to use only once the session_start() at the entry point of the site on the main top of the script and destroy the actual session whenever a use want to logout
- Its possible that something  like user profile will be deleted its not very an urgent task to do there way more important things
- Exceptions blocks are now centralized at then entry point of the site so each case have his own 

PS : Like I said in one of the commit regardless to exceptions it's possible that some changes come in the nextdays

### CODE REVIEW

File: src/Controller/ArticleController.php [hunk 1]
```diff
@@ -17,10 +28,124 @@ public function listOfAllArticles(): array
...
+  }
+
+
+
+  public function handleTitleField(string $title): array|string
+  {
+    $titleRegex = "/^(?=.{1,51}$)[A-ZÀ-ÿ][A-Za-zÀ-ÿ -']*$/";
+    switch (true) {
+      case empty($title):
+        header("HTTP/1.1 400");
```
Comment (james2001) [inline src/Controller/ArticleController.php]: don't send the header("HTTP/1.1 400");

File: src/Controller/ArticleController.php [hunk 2]
```diff
@@ -17,10 +28,124 @@ public function listOfAllArticles(): array
...
+      case !preg_match($tagsRegex, $tags):
+        header("HTTP/1.1 400");
+        throw new TagsWrongFormatException();
+      default:
+        return ["tags" => $tags];
+    }
+  }
+
+
+  public function handleCreateArticleValidator(string $title, array $fileArticle, string $shortPhrase, string $content, string $tags, array $sessionData): void
```
Comment (james2001) [inline src/Controller/ArticleController.php]: Il faudrait améliorer la gestion d'erreur

File: src/Model/SessionManager.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Model;
+
+class SessionManager
+{
+
+  public function startSession(): void
+  {
+    if (session_status() !== PHP_SESSION_ACTIVE) session_start();
```
Comment (james2001) [inline src/Model/SessionManager.php]:
```suggestion
    if (session_status() !== PHP_SESSION_ACTIVE) {
     session_start();
     }
```

File: vendor/composer/autoload_psr4.php [hunk 4]
```diff
@@ -11,8 +11,8 @@
     'Symfony\\Polyfill\\Ctype\\' => array($vendorDir . '/symfony/polyfill-ctype'),
     'PHPMailer\\PHPMailer\\' => array($vendorDir . '/phpmailer/phpmailer/src'),
     'Model\\' => array($baseDir . '/src/Model'),
-    'Exceptions\\' => array($baseDir . '/src/Class/Exceptions'),
```
Comment (james2001) [inline vendor/composer/autoload_psr4.php:14]: Ne pas commit les fichier les vendor

File: src/Controller/ArticleController.php (hunk 2 above)
Comment (Jexinte) [reply src/Controller/ArticleController.php]: Done

File: src/Model/SessionManager.php (hunk 3 above)
Comment (Jexinte) [reply src/Model/SessionManager.php]: Done

File: src/Controller/ArticleController.php (hunk 1 above)
Comment (Jexinte) [reply src/Controller/ArticleController.php]: Correction done

File: vendor/composer/autoload_psr4.php (hunk 4 above)
Comment (Jexinte) [reply vendor/composer/autoload_psr4.php:14]: Noted

**Pull request merged**